db/data/syllabusesModel.js

```diff
@@ -8,7 +8,10 @@ const MONGO_HOSTNAME = process.env.MONGO_HOSTNAME || 'localhost';
 const MONGO_PORT = process.env.MONGO_PORT || 27017;
 const MONGO_DB = process.env.MONGO_DB || 'syllabuses';
 
-mongoose.connect(`mongodb://${MONGO_HOSTNAME}:${MONGO_PORT}/${MONGO_DB}`, { useNewUrlParser: true, useUnifiedTopology: true });
+const connect = () => new Promise((resolve, reject) => {
+  mongoose.connect(`mongodb://${MONGO_HOSTNAME}:${MONGO_PORT}/${MONGO_DB}`, { useNewUrlParser: true, useUnifiedTopology: true });

```

Comment:
I'm wondering if you have already populated and started to use this particular db/collection? I

'm asking because the env vars are exposed on github (might not matter at this point for you), and I'm guessing the connection string connects to **localhost**, and to a db named **syllabuses**?

Reply:
@phucciTr - You're right, yeah. This is legacy code, I think the original author did it this way to be able to make quick switches between development and production. In my case, I'm not using a `.env` file anyway, so these env variables aren't being used right now.